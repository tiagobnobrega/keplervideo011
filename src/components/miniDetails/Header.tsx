import { isEqual } from 'lodash';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { COLORS } from '../../styles/Colors';
import { TitleData } from '../../types/TitleData';
import { scaleUxToDp } from '../../utils/pixelUtils';

interface HeaderProps {
  data?: TitleData;
}

const Header = ({ data }: HeaderProps) => {
  return (
    <View style={styles.headerContent}>
      <Text style={styles.titleText} numberOfLines={2}>
        {data?.title}
      </Text>
      <Text style={styles.descriptionText} numberOfLines={3}>
        {data?.description}
      </Text>
    </View>
  );
};

const headerDataUnchanged = (
  prevProps: HeaderProps,
  nextProps: HeaderProps,
) => {
  return isEqual(prevProps.data, nextProps.data);
};

export default React.memo(Header, headerDataUnchanged);

const styles = StyleSheet.create({
  headerContent: {
    paddingStart: scaleUxToDp(120),
    paddingTop: scaleUxToDp(138),
    width: scaleUxToDp(884),
  },
  titleText: {
    fontSize: scaleUxToDp(60),
    fontWeight: 'bold',
    color: COLORS.SMOKE_WHITE,
    width: scaleUxToDp(800),
  },
  descriptionText: {
    fontSize: scaleUxToDp(30),
    color: COLORS.SMOKE_WHITE,
    width: scaleUxToDp(884),
  },
});
