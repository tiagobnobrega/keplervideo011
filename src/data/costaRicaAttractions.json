[
  {
    "id": "169304",
    "title": "4k 60 FPS Test Video",
    "description": "Trip to Costa Rica 2024 Highlight",
    "duration": 387,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "videoUrl": "https://storage.googleapis.com/exoplayer-test-media-1/60fps/bbb-clear-2160/manifest.mpd",
    "categories": ["Costa Rica Islands"],
    "channelID": "13090",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "928272",
    "title": "Dolby 1",
    "description": "Dolby H264 and EC3 DASH stream with fMP4 segment(s)",
    "duration": 387,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "videoUrl": "https://ott.dolby.com/OnDelKits/DDP/Dolby_Digital_Plus_Online_Delivery_Kit_v1.5/Test_Signals/muxed_streams/DASH/OnDemand/MPD/Audio_ID_720p_25fps_h264_6ch_256kbps_ddp.mpd",
    "categories": ["Costa Rica Islands"],
    "channelID": "13090",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "137363",
    "title": "Dolby 2",
    "description": "Dolby H265 and EC3 DASH stream with fMP4 segment(s)",
    "duration": 387,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "videoUrl": "https://ott.dolby.com/OnDelKits/DDP/Dolby_Digital_Plus_Online_Delivery_Kit_v1.5/Test_Signals/muxed_streams/DASH/OnDemand/MPD/Audio_ID_720p_25fps_h265_6ch_256kbps_ddp.mpd",
    "categories": ["Costa Rica Islands"],
    "channelID": "13090",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "HLS",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "169309",
    "title": "Dolby 3",
    "description": "Dolby H264 and EC3 HLS stream with TS segment(s)",
    "duration": 387,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "videoUrl": "https://ott.dolby.com/OnDelKits/DDP/Dolby_Digital_Plus_Online_Delivery_Kit_v1.4.1/Test_Signals/muxed_streams/HLS/Manifest_v3/ChID_voices_1280x720p_25fps_h264_6ch_256kbps_ddp.m3u8",
    "categories": ["Costa Rica Islands"],
    "channelID": "13090",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "HLS",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "169304",
    "title": "Dolby 4",
    "description": "Dolby H264 and EC3 HLS stream",
    "duration": 387,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "videoUrl": "https://ott.dolby.com/OnDelKits/DDP/Dolby_Digital_Plus_Online_Delivery_Kit_v1.4.1/Test_Signals/muxed_streams/HLS/Manifest_v3/ChID_voices_1280x720p_25fps_h264_6ch_256kbps_ddp.m3u8",
    "categories": ["Costa Rica Islands"],
    "channelID": "13090",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "169313",
    "title": "Beautiful Whale Tail Uvita Costa Rica",
    "description": "Beautiful Whale Tail Uvita Costa Rica",
    "duration": 86,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169313/video_1880k/T7J66Z106.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169309",
    "title": "Tears",
    "description": "MPD drm format Tears description",
    "duration": 130,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169309/thumbs/0169309__009f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169309/thumbs/0169309__009f.jpg",
    "videoUrl": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD_DRM",
    "drmScheme": {
      "name": "com.widevine.alpha",
      "licenseUri": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
    },
    "secure": true,
    "uhd": false,
    "textTrack": [
      {
        "label": "English",
        "language": "en",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-en.vtt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ],
    "rentAmount": "135"
  },
  {
    "id": "169312",
    "title": "Tears of Steel",
    "description": "MPD format Tears of Steel",
    "duration": 192,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169312/thumbs/0169312__003f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169312/thumbs/0169312__003f.jpg",
    "videoUrl": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/manifest.mpd",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD",
    "secure": false,
    "uhd": false,
    "textTrack": [
      {
        "label": "English [en]",
        "language": "en",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-en.vtt",
        "mimeType": "application/x-subtitle-vtt"
      },
      {
        "label": "German [de]",
        "language": "de",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-de.vtt",
        "mimeType": "application/x-subtitle-vtt"
      },
      {
        "label": "French [fr]",
        "language": "fr",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-fr.vtt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ],
    "rentAmount": "135"
  },
  {
    "id": "169322",
    "title": "Big Buck Bunny: The Dark Truths",
    "description": "HLS format Big Buck Bunny: The Dark Truths",
    "duration": 205,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169322/thumbs/0169322__002f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169322/thumbs/0169322__002f.jpg",
    "videoUrl": "https://storage.googleapis.com/shaka-demo-assets/bbb-dark-truths-hls/hls.m3u8",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "HLS",
    "secure": false,
    "uhd": false,
    "textTrack": [
      {
        "label": "Sample CC 1",
        "language": "en",
        "uri": "https://mtoczko.github.io/hls-test-streams/test-vtt-ts-segments/text/1.vtt",
        "mimeType": "application/x-subtitle-vtt"
      },
      {
        "label": "Sample CC 2",
        "language": "es",
        "uri": "https://brenopolanski.github.io/html5-video-webvtt-example/MIB2-subtitles-pt-BR.vtt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ],
    "rentAmount": "135"
  },
  {
    "id": "169322",
    "title": "Scuba Diving - Costa Rica",
    "description": "Scuba Diving - Costa Rica (Playa Ocotal & The Bat Islands)",
    "duration": 205,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169322/thumbs/0169322__002f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169322/thumbs/0169322__002f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169322/video_1880k/S6IZ6M1QM.mp4?source=firetv&channelID=13455",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169312",
    "title": "San Lucas Trip",
    "description": "Isla San Lucas, Puntarenas, Costa Rica",
    "duration": 192,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169312/thumbs/0169312__003f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169312/thumbs/0169312__003f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169312/video_1880k/M0CAAG18G.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169309",
    "title": "San Jose, Costa Rica",
    "description": "San Jose, Costa Rica",
    "duration": 130,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169309/thumbs/0169309__009f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169309/thumbs/0169309__009f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169309/video_1880k/88HFXX0IL.mp4?source=firetv&channelID=13453",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169321",
    "title": "Costa Rica Diving - Bat Island",
    "description": "Costa Rica Diving - Bat Island",
    "duration": 169,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169321/thumbs/0169321__002f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169321/thumbs/0169321__002f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169321/video_1880k/L9BGAF0JX.mp4?source=firetv&channelID=13455",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169311",
    "title": "Tortuguero, Costa Rica",
    "description": "Tortuguero, Costa Rica",
    "duration": 642,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169311/thumbs/0169311__013f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169311/thumbs/0169311__013f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169311/video_1880k/62XJJI1MA.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169308",
    "title": "Pos Volcano, Costa Rica Timelapse",
    "description": "Pos Volcano, Costa Rica Timelapse",
    "duration": 45,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169308/thumbs/0169308__011f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169308/thumbs/0169308__011f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169308/video_1880k/M0CAGG18M.mp4?source=firetv&channelID=13453",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  }
]
