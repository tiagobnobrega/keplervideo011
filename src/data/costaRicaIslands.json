[
  {
    "id": "169304",
    "title": "Trip to Costa Rica 2016 Highlight",
    "description": "Trip to Costa Rica 2016 Highlight",
    "duration": 387,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169304/thumbs/0169304__005f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169304/video_1880k/U88P661SQ.mp4?source=firetv&channelID=13453",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "133421",
    "title": "Uvita, Dominical, Manuel Antonio, playa Ventanas, Costa Rica",
    "description": "Hey Everyone,,,, if you find yourself down in the Southern Pacific Zone of Costa Rica....I think you might want to check Playa Ventanas OUT.... Way COOL beach,,rustic and natural, off the beaten path, just what you want when you don't want the tourist thing. Trust us...... it's worth a look.",
    "duration": 96,
    "thumbnail": "http://le2.cdn01.net/videos/0000133/0133421/thumbs/0133421__006f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000133/0133421/thumbs/0133421__006f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000133/0133421/video_1880k/X8HCGX0JD.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "169315",
    "title": "Discover Cocos Island",
    "description": "Join us for an expedition to the Worlds most beautiful bbig-animal destination; Cocos Island.",
    "duration": 151,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169315/thumbs/0169315__005f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169315/thumbs/0169315__005f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169315/video_1880k/G4OBBH0XD.mp4?source=firetv&channelID=13455",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": false,
    "rentAmount": "135"
  },
  {
    "id": "133420",
    "title": "Uvita Costa Rica La Cascada el Pavon",
    "description": "Hey Everyone,,,Travel a little south of Uvita and across from Playa Ocochal (Ojochal) take a left at the sign advertising the Tilapa El Pavon Restaurant. On the way to the restaurant (which we have a video about and also recommend) on the left hand side you should see the sign for La Cascada el Pavon (Turkey Waterfall). We highly suggest you stop and take the very short hike down to the waterfall and check it out. Way COOL! Dont take our word for it- Check out this video. Pura Vida!",
    "duration": 105,
    "thumbnail": "http://le1.cdn01.net/videos/0000133/0133420/thumbs/0133420__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000133/0133420/thumbs/0133420__005f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000133/0133420/video_1880k/G0CXIG1LX.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  }
]
