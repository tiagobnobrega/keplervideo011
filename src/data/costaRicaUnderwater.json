[
  {
    "id": "169319",
    "title": "Diving Costa Rica 2017",
    "description": "Diving Costa Rica 2017",
    "duration": 446,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169319/thumbs/0169319__010f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169319/thumbs/0169319__010f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169319/video_1880k/A57CCB0FI.mp4?source=firetv&channelID=13455",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169310",
    "title": "Samara, Costa Rica",
    "description": "Samara, Costa Rica",
    "duration": 594,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169310/thumbs/0169310__015f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169310/thumbs/0169310__015f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169310/video_1880k/Q4GLL816I.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169306",
    "title": "Costa Rica Tourism 2016",
    "description": "Costa Rica Tourism 2016",
    "duration": 3548,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169306/thumbs/0169306__005f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169306/thumbs/0169306__005f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169310/video_1880k/Q4GLL816I.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169317",
    "title": "Diving Costa Rica - Catalina Islands",
    "description": "Diving Costa Rica - Catalina Islands",
    "duration": 227,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169317/thumbs/0169317__015f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169317/thumbs/0169317__015f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169317/video_1880k/L9BGGF0J8.mp4?source=firetv&channelID=13455",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169305",
    "title": "Costa Rica 2016",
    "description": "Our vacation in Costa Rica for a week during May/June 2016.",
    "duration": 501,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169305/thumbs/0169305__010f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169305/thumbs/0169305__010f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169305/video_1880k/J79XDD0AG.mp4?source=firetv&channelID=13453",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "133422",
    "title": "Costa Rica Heliconia Island",
    "description": "We went to Costa Rica in 2010. We travelled around by a Hired car 4x4. We started in Dos Palmas near Alejuela. Our first thereafter was Mawamba Lodge then Heliconian Island, Peace Lodge,Linda vista, Ceiba Lodge,Blue Dream hotel, Elvera hotel Coco Palms, Miramonte Swiss,Casino Hotel, Trogon Lodge, Dos Palmas. All together about 2500 km. We found our way by using the Garmin Gps 60csx and the topo from Costa Rica. This worked out amazingly well.",
    "duration": 268,
    "thumbnail": "http://le1.cdn01.net/videos/0000133/0133422/thumbs/0133422__004f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000133/0133422/thumbs/0133422__004f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000133/0133422/video_1880k/L5AJZL1QL.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  },
  {
    "id": "169316",
    "title": "Costa Rica underwater",
    "description": "Costa Rica underwater",
    "duration": 480,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169316/thumbs/0169316__007f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169316/thumbs/0169316__007f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169316/video_1880k/Z1DIIA1LL.mp4?source=firetv&channelID=13455",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "rentAmount": "135"
  }
]
