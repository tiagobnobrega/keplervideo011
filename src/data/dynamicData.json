[
  {
    "id": "169313",
    "title": "mp4 Test Footage",
    "description": "mp4 Test Footage",
    "duration": 86,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg",
    "videoUrl": "https://edge-vod-media.cdn01.net/encoded/0000169/0169313/video_1880k/T7J66Z106.mp4?source=firetv&channelID=13454",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MP4",
    "secure": false,
    "uhd": true,
    "textTrack": [
      {
        "label": "Label",
        "language": "en",
        "uri": "https://mtoczko.github.io/hls-test-streams/test-vtt-ts-segments/text/1.vtt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ]
  },
  {
    "id": "169322",
    "title": "Big Buck Bunny: The Dark Truths",
    "description": "HLS format Big Buck Bunny: The Dark Truths",
    "duration": 205,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169322/thumbs/0169322__002f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169322/thumbs/0169322__002f.jpg",
    "videoUrl": "https://storage.googleapis.com/shaka-demo-assets/bbb-dark-truths-hls/hls.m3u8",
    "categories": ["Costa Rica Underwater"],
    "channelID": "13455",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "HLS",
    "secure": false,
    "uhd": false,
    "textTrack": [
      {
        "label": "Sample CC 1",
        "language": "en",
        "uri": "https://mtoczko.github.io/hls-test-streams/test-vtt-ts-segments/text/1.vtt",
        "mimeType": "application/x-subtitle-vtt"
      },
      {
        "label": "Sample CC 2",
        "language": "es",
        "uri": "https://brenopolanski.github.io/html5-video-webvtt-example/MIB2-subtitles-pt-BR.vtt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ]
  },
  {
    "id": "169312",
    "title": "Tears of Steel",
    "description": "MPD format Tears of Steel",
    "duration": 192,
    "thumbnail": "http://le1.cdn01.net/videos/0000169/0169312/thumbs/0169312__003f.jpg",
    "posterUrl": "http://le1.cdn01.net/videos/0000169/0169312/thumbs/0169312__003f.jpg",
    "videoUrl": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/manifest.mpd",
    "categories": ["Costa Rica Islands"],
    "channelID": "13454",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD",
    "secure": false,
    "uhd": false,
    "textTrack": [
      {
        "label": "English [en]",
        "language": "en",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-en.vtt",
        "mimeType": "application/x-subtitle-vtt"
      },
      {
        "label": "German [de]",
        "language": "de",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-de.vtt",
        "mimeType": "application/x-subtitle-vtt"
      },
      {
        "label": "French [fr]",
        "language": "fr",
        "uri": "https://media.axprod.net/TestVectors/Cmaf/clear_1080p_h264/subtitle-fr.vtt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ]
  },
  {
    "id": "169309",
    "title": "Tears",
    "description": "MPD drm format Tears description",
    "duration": 130,
    "thumbnail": "http://le2.cdn01.net/videos/0000169/0169309/thumbs/0169309__009f.jpg",
    "posterUrl": "http://le2.cdn01.net/videos/0000169/0169309/thumbs/0169309__009f.jpg",
    "videoUrl": "https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd",
    "categories": ["Costa Rica Attractions"],
    "channelID": "13453",
    "rating": "2.5",
    "mediaType": "video",
    "mediaSourceType": "url",
    "format": "MPD",
    "drmScheme": {
      "name": "com.widevine.alpha",
      "licenseUri": "https://proxy.uat.widevine.com/proxy?provider=widevine_test"
    },
    "secure": true,
    "uhd": false,
    "textTrack": [
      {
        "label": "English",
        "language": "en",
        "uri": "https://download.blender.org/demo/movies/ToS/subtitles/TOS-en.srt",
        "mimeType": "application/x-subtitle-vtt"
      }
    ]
  }
]
