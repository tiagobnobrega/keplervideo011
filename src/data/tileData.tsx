import { TitleData } from '../types/TitleData';

export const tileData: TitleData = {
  id: '169313',
  title: 'Beautiful Whale Tail Uvita Costa Rica',
  description: 'Beautiful Whale Tail Uvita Costa Rica',
  duration: 86,
  thumbnail:
    'http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg',
  posterUrl:
    'http://le2.cdn01.net/videos/0000169/0169313/thumbs/0169313__007f.jpg',
  videoUrl:
    'https://edge-vod-media.cdn01.net/encoded/0000169/0169313/video_1880k/T7J66Z106.mp4?source=firetv&channelID=13454',
  categories: ['Costa Rica Islands'],
  channelID: '13454',
  mediaType: 'video',
  mediaSourceType: 'url',
  format: 'MP4',
  secure: false,
  uhd: false,
  rentAmount: '100',
} as TitleData;
