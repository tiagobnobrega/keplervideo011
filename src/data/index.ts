import { RotatorData } from '../components/rotator/type';
import { TitleData } from '../types/TitleData';

export const TestData: () => TitleData[] = () => require('./dynamicData.json');
export const CostaRicaAttractionsData: () => TitleData[] = () =>
  require('./costaRicaAttractions.json');
export const CostaRicaIslandsData: () => TitleData[] = () =>
  require('./costaRicaIslands.json');
export const CostaRicaUnderwaterData: () => TitleData[] = () =>
  require('./costaRicaUnderwater.json');
export const CostaRicaTopRatedData: () => TitleData[] = () =>
  require('./costaRicaTopRated.json');
export const AutoRotatorData: RotatorData[] = require('./rotatorData.json');
