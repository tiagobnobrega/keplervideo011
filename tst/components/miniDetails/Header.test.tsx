import { render } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import Header from '../../../src/components/miniDetails/Header';
import { tileData } from '../../../src/data/tileData';

describe('Header renders correctly', () => {
  it('renders correctly without styles', () => {
    const tree = render(<Header />);
    expect(tree).toMatchSnapshot();
  });
});
describe('Header with React.memo', () => {
  it('renders only when props change', async () => {
    const tileDataTwo = { ...tileData, id: '169314' };
    const isEqualSpy = jest.spyOn(require('lodash'), 'isEqual');
    const { rerender, UNSAFE_getByProps } = render(<Header data={tileData} />);
    rerender(<Header data={tileData} />);
    expect(isEqualSpy).toHaveBeenCalledWith(tileData, tileData);
    rerender(<Header data={tileDataTwo} />);
    expect(isEqualSpy).toHaveBeenCalledWith(tileData, tileDataTwo);
    expect(UNSAFE_getByProps({ data: tileDataTwo })).toBeTruthy();
  });
});
