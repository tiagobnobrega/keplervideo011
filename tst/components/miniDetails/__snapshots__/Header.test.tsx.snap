// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly renders correctly without styles 1`] = `
<View
  style={
    Object {
      "paddingStart": 120,
      "paddingTop": 138,
      "width": 884,
    }
  }
>
  <Text
    numberOfLines={2}
    style={
      Object {
        "color": "#F1F1F1",
        "fontSize": 60,
        "fontWeight": "bold",
        "width": 800,
      }
    }
  />
  <Text
    numberOfLines={3}
    style={
      Object {
        "color": "#F1F1F1",
        "fontSize": 30,
        "width": 884,
      }
    }
  />
</View>
`;
